src/sage/schemes/hyperelliptic_curves/hypellfrob.pyx

```diff
@@ -175,24 +162,17 @@ def hypellfrob(p, N, Q):
     INPUT:
 
     - ``p`` -- a prime
-    - ``Q`` -- a monic polynomial in `\ZZ[x]` of odd degree.
-      Must have no multiple roots mod `p`.
-    - ``N`` -- precision parameter; the output matrix will be correct modulo `p^N`.
-
-    PRECONDITIONS:
+    - ``Q`` -- a monic polynomial in `\ZZ[x]` of odd degree; must have no
+      multiple roots mod `p`.
+    - ``N`` -- precision parameter; the output matrix will be correct modulo `p^N`
 
-    Must have `p > (2g+1)(2N-1)`, where `g = (\deg(Q)-1)/2` is the genus
-    of the curve.
+    The prime `p` should satisfy `p > (2g+1)(2N-1)`, where `g =
+    \frac{\deg(Q)-1}{2}` is the genus of the curve.
 
     ALGORITHM:
 
-    Described in "Kedlaya's algorithm in larger characteristic" by David
-    Harvey. Running time is theoretically soft-`O(p^{1/2} N^{5/2} g^3)`.
-
-    .. TODO::
-
-        Remove the restriction on `p`. Probably by merging in Robert's code,
-        which eventually needs a fast C++/NTL implementation.
+    Described in [Harv2007]_. Running time is theoretically soft-`O(p^{1/2}
```

Comment:
```suggestion
    Described in [Harv2007]_, section 7. Running time is theoretically `\widetilde{O}(p^{1/2}
```
$\text{soft-}O \mapsto \widetilde{O}$

Reply:
I thought that, but was not sure. Thanks.